CMakeLists.txt

```
@@ -87,6 +87,11 @@ if(${USE_Z3_DLOPEN})
 elseif (${Z3_FOUND})
   add_definitions(-DHAVE_Z3)
   message("Z3 SMT solver found. This enables optional SMT checking with Z3.")
+  if (TESTS AND NOT EMSCRIPTEN)
+    if ("${Z3_VERSION}" VERSION_LESS "4.8.12.0")
+      message(WARNING "SMT tests need Z3 4.8.12 or higher, SMT tests are expected to fail!")
```

Could you add a variable/constant for the required Z3 version?
```suggestion
    if ("${Z3_VERSION}" VERSION_LESS ${LATEST_Z3_VERSION})
      message(WARNING "SMT tests need Z3 ${LATEST_Z3_VERSION}, SMT tests are expected to fail!")
```

---

done